From edc40b476f0e779522bf6e4b22cd36451e6b021a Mon Sep 17 00:00:00 2001
From: Catalin Marinas <catalin.marinas@arm.com>
Date: Tue, 18 Apr 2017 16:32:26 +0800
Subject: [PATCH 263/767] usb: dwc3: initiated xhci probe problem in arm64 4.4
 kernel due to DMA setup

Signed-off-by: Catalin Marinas <catalin.marinas@arm.com>
Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/usb/dwc3/host.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/host.c b/drivers/usb/dwc3/host.c
index 1a3878a3be78..aaaa37193546 100644
--- a/drivers/usb/dwc3/host.c
+++ b/drivers/usb/dwc3/host.c
@@ -9,6 +9,8 @@
 
 #include <linux/platform_device.h>
 
+#include <linux/of_device.h>
+
 #include "core.h"
 
 static int dwc3_host_get_irq(struct dwc3 *dwc)
@@ -77,7 +79,14 @@ int dwc3_host_init(struct dwc3 *dwc)
 		return -ENOMEM;
 	}
 
-	xhci->dev.parent	= dwc->dev;
+	if (IS_ENABLED(CONFIG_OF) && dwc->dev->of_node)
+		of_dma_configure(&xhci->dev, dwc->dev->of_node);
+	else
+		dma_set_coherent_mask(&xhci->dev, dwc->dev->coherent_dma_mask);
+
+	xhci->dev.parent = dwc->dev;
+
+	xhci->dev.dma_mask = dwc->dev->dma_mask;
 
 	dwc->xhci = xhci;
 
-- 
2.17.0

